pandas/_libs/window.pyx

```diff
@@ -442,80 +182,75 @@ cdef inline void remove_sum(float64_t val, int64_t *nobs, float64_t *sum_x) nogi
         sum_x[0] = sum_x[0] - val
 
 
-def roll_sum(ndarray[float64_t] values, int64_t win, int64_t minp,
-             object index, object closed):
+def roll_sum_variable(ndarray[float64_t] values, ndarray[int64_t] start,
```

Comment:
does `ndarray[type_t]` vs `type_t[:]` make a difference here?

Reply:
Not entirely noticeable?

```
# np buffer
In [1]: N = 1_000_000
   ...: s = pd.Series(range(N), index=pd.date_range('2019', periods=N, freq='s'))
   ...: roll = s.rolling('1H')
   ...: %timeit roll.sum()
28.8 ms ± 486 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

# memoryview
In [1]: N = 1_000_000
   ...: s = pd.Series(range(N), index=pd.date_range('2019', periods=N, freq='s'))
   ...: roll = s.rolling('1H')
   ...: %timeit roll.sum()
28.8 ms ± 416 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```